### PR TITLE
skip not found when collecting logtail

### PR DESCRIPTION
**Pull request by aptend**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6157

## What this PR does / why we need it:

we should skip not found dirty blocks (they are compacted) when collecting logtail

### CODE REVIEW

File: pkg/vm/engine/tae/logtail/operator.go [hunk 1]
```diff
@@ -59,14 +60,22 @@ func (op *BoundOperator) Run() (err error) {
 		}
 		for _, dirtySeg := range tblDirty.Segs {
 			if seg, err = tbl.GetSegmentByID(dirtySeg.ID); err != nil {
-				return err
+				if moerr.IsMoErrCode(err, moerr.ErrNotFound) {
```
Comment (fengttt) [inline pkg/vm/engine/tae/logtail/operator.go:63]: What is the frequency of this not found?   If it is really often, please considered an Ok...EOB/END code.   Basically we if not found is the common case, we want an Ok code.

Comment (aptend) [reply pkg/vm/engine/tae/logtail/operator.go:63]: the frequency is related to the flush interval option. GetSegmentByID etc. are not just only used in collecting logtail, so here it is better to ignore the not found err and do not change the code returned by GetSegmentByID etc

Comment (XuPeng-SH) [reply pkg/vm/engine/tae/logtail/operator.go:63]: ErrNotFound is replaced with GetOK.... 
It should be modified when #6166 is merged into the main.

**Review by XuPeng-SH — CHANGES_REQUESTED** [review]
ErrNotFound is replaced by GetOKXXXX.
It should be modified when #6166 is merged into the main.

**Review by fengttt — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Pull request merged**